core/dbt/cli/context.py

```diff
@@ -0,0 +1,39 @@
+from click import Context, Group, Command
+from click.exceptions import NoSuchOption, UsageError
+from dbt.config.runtime import load_project, load_profile
+from dbt.cli.flags import Flags
+import sys
+
+
+class DBTUsageException(Exception):
+    pass
+
+
+class DBTContext(Context):
+    def __init__(self, command: Command, **kwargs) -> None:
+        invocation_args = kwargs.pop("args", sys.argv[1:])
```

Comment:
I think we probably can't use sys.argv here? For anything invoking it programmatically

Reply:
agree, still need to fix this up 👍